### PR TITLE
Batch of GHA CI updates

### PR DESCRIPTION
**Pull request by dfarrell07**
See commit messages for details.

### CODE REVIEW

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -1,18 +1,26 @@
...
     steps:
-      - name: Checkout the repository
-        uses: actions/checkout@master
+      - name: Check out the repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Fetch git tags
+        run: git fetch origin +refs/tags/*:refs/tags/*
```
Comment (mkolesnik) [inline .github/workflows/e2e.yaml:23]: Why is this actually necessary?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 jobs:
   release:
+    name: Release
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@master
+      - name: Check out the repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (mkolesnik) [inline .github/workflows/release.yml:19]: Same here, why is this necessary?

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (dfarrell07) [reply .github/workflows/e2e.yaml:23]: It might not be, I will check. We do the same in the Operator, so if it's not necessary I guess we should remove it there as well.

File: .github/workflows/release.yml (hunk 2 above)
Comment (dfarrell07) [reply .github/workflows/release.yml]: Removed, thanks.

File: .github/workflows/e2e.yaml (hunk 1 above)
Comment (dfarrell07) [reply .github/workflows/e2e.yaml]: See also https://github.com/submariner-io/submariner-operator/pull/762.

Comment (mkolesnik) [reply .github/workflows/e2e.yaml]: On the operator it's actually necessary since otherwise the CI can't calculate the version, but anywhere else it's not necessary to clone with depth 0 or to fetch the tags, hence my comment.

Perhaps in the operator it's better to comment about this in the GHA file so it's clearer why it's done.

Comment (dfarrell07) [reply .github/workflows/e2e.yaml]: @mkolesnik I think it's necessary in some places in the Operator, but not in the locations removed in https://github.com/submariner-io/submariner-operator/pull/762. Do you think those are necessary too?

**Review by mkolesnik — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]
